ext/symengine/ruby_function.c

```diff
@@ -37,3 +39,39 @@ IMPLEMENT_ONE_ARG_FUNC(gamma);
 
 #undef IMPLEMENT_ONE_ARG_FUNC
 
+VALUE cfunction_functionsymbol_init(VALUE self, VALUE args)
+{
+    int argc = NUM2INT(rb_funcall(args, rb_intern("length"), 0, NULL));
```

Comment:
Or you can use [`rb_ary_length`](http://rxr.whitequark.org/mri/source/array.c#1766).